charon/pkgs/signature.py

```diff
@@ -0,0 +1,152 @@
+"""
+Copyright (C) 2022 Red Hat, Inc. (https://github.com/Commonjava/charon)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import subprocess
+import asyncio
+import logging
+import gnupg
+from typing import Awaitable, Callable, List, Tuple
+from charon.storage import S3Client
+
+logger = logging.getLogger(__name__)
+
+
+def generate_sign(
+    package_type: str,
+    artifact_path: List[str],
+    top_level: str,
+    prefix: str,
+    s3_client: S3Client,
+    bucket: str,
+    key_id: str = None,
+    key_file: str = None,
+    passphrase: str = None
+) -> Tuple[List[str], List[str]]:
```

Comment:
Can you give some api doc for this method to show what purpose of it? Like the param meaning and the process purpose?

Reply:
K, I'll update it